cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -346,6 +367,15 @@ func isInstanceNotRunningYet(gceInstance *gce.ManagedInstance) bool {
 	return gceInstance.InstanceStatus == "" || gceInstance.InstanceStatus == "PROVISIONING" || gceInstance.InstanceStatus == "STAGING"
 }
 
+func generateInstanceName(migRef GceRef, existingNames map[string]bool) string {
+	for {
```

Comment:
WDYT about adding some condition to this loop just to make sure it doesn't loop forever (e.g. for i=0;i<100;i++ and maybe a warning after the loop)? I might be too paranoid though :stuck_out_tongue:

Reply:
Added condition and warning after loop